scisim/static/js/Utils/LogManager.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Specific log manager for logging choices on a page
+ */
+var ChoiceLogManager = function() {
+    this.choices = [];
+    this.stopwatch = new StopWatch();
+};
+
+ChoiceLogManager.prototype.logChoice = function(choiceInfo) {
+    this.choices.push(choiceInfo);
+
+    var loggableString = "";
+
+    if(choiceInfo.page_context.hasOwnProperty("patient")){
+        loggableString += choiceInfo.page_context.patient.name + ": ";
+    }else{
+        loggableString += "Question: " + choiceInfo.prev.text() + " You said: ";
+    }
+
+    loggableString += choiceInfo.choice;
+
+    labnotebook.add(loggableString.replace("}", ""));
+
+    if (choiceInfo.page_context.hasOwnProperty("patient")) {
+        choiceInfo.page_context.patient.choices.push(loggableString);
```

Comment:
note that when you update a context, you have to call chain.updateContext on the context.